rules/appinsights-authentication/rule.md

```diff
@@ -0,0 +1,194 @@
+---
+type: rule
+archivedreason: 
+title: Do you secure your AppInsights Telemetry?
+guid: f2155eb7-6ade-4757-b962-a1dc680cdbbf
+uri: appinsights-authentication
+created: 2023-10-16T15:11:34.0000000Z
+authors:
+- title: William Liebenberg
+  url: https://ssw.com.au/people/william-liebenberg
+- title: Alex Rothwell
+  url: https://www.ssw.com.au/people/alex-rothwell/
+related: []
+redirects: []
+
+---
+
+Azure Application Insights is your app's friendly doctor, keeping an eye on it to stay healthy and running smoothly. It uses an `Instrumentation Key` to allow your app to send telemetry data to its resource.
+
+While this `Instrumentation Key` isn't a secret and is often included in client-side code, it's crucial to keep it safe. If it falls into the wrong hands, it could corrupt your data or lead to extra costs.
+
+The key doesn't grant access to your data; for that, you use Entra ID (used to be Azure AD) authentication.
+
+To preventing unintended access, Local authentication (typically Instrumentation Key or API Keys) can be **disabled** for authenticating data ingestion. Once disabled, a **stronger authentication** method known as Role **Based Access Control (RBAC)** can be used.
```

Comment:
This is a great rule, but I can see this rule can be used in the unintended direction.

I think we should clearly outline the trade-off of securing application insights.
Enabling this effectively blocks sending any client-side telemetries to this AI instance - this approach will require us to do an alternative such as:
- create another application insight and then combine them together somehow
- only use SSR frontend
- create a secure ingestion point in backend (I think this is a bad approach)
- secure frontend with user AD Authentication (locks us down to AD)

What do you think? 😄

Reply:
I agree with everything you listed above. There will be trade offs.

I can add the PROs and CONs for securing server-side telemetry, but I'll dig a bit further to see what we can recommend for client-side telemetry.

Thanks @christoment !